src/app/src/pages/redteam/setup/components/StrategyConfigDialog.tsx

```diff
@@ -55,6 +70,85 @@ export default function StrategyConfigDialog({
   const [goals, setGoals] = React.useState<string[]>(config.goals || []);
   const [newGoal, setNewGoal] = React.useState<string>('');
 
+  // Steps are just strategy IDs (strings)
+  const [steps, setSteps] = React.useState<string[]>(config.steps || []);
+  const [newStep, setNewStep] = React.useState<string>('');
+  const [layerPlugins, setLayerPlugins] = React.useState<string[]>(config.plugins || []);
+  // Plugin targeting: 'all' or 'specific'
+  const [pluginTargeting, setPluginTargeting] = React.useState<'all' | 'specific'>(
+    config.plugins && config.plugins.length > 0 ? 'specific' : 'all',
+  );
+
+  // Filter available plugins to only show those selected in the Plugins step
+  const availablePlugins = React.useMemo(() => {
+    if (selectedPlugins.length === 0) {
+      return ALL_PLUGINS;
+    }
+    return ALL_PLUGINS.filter((plugin) => selectedPlugins.includes(plugin));
+  }, [selectedPlugins]);
```

Comment:
?? Why not just return selectedPlugins? why would you wanna show all if the user didn't select any from the previous step?